src/OpenTelemetry/Trace/Export/BatchingActivityProcessor.cs

```diff
@@ -140,12 +140,7 @@ public override async Task ShutdownAsync(CancellationToken cancellationToken)
                 this.cts.Dispose();
                 this.cts = null;
 
-                // if there are more items, continue until cancellation token allows
-                while (this.currentQueueSize > 0 && !cancellationToken.IsCancellationRequested)
-                {
-                    await this.ExportBatchAsync(cancellationToken).ConfigureAwait(false);
-                }
-
+                await this.ForceFlushAsync(cancellationToken);
                 await this.exporter.ShutdownAsync(cancellationToken);
```

Comment:
Not from your change and a separate discussion likely not to be resolved on this PR:

1. Code should check if a cancellation was requested before calling `exporter.ShutdownAsync`;
2. It is a bit strange that the processor itself is in charge of calling the shutdown on the exporter;